File: frontend/src/types/audio.ts
export interface FrequencyFeatures {
  fundamental: number;
  spectrum: number[];
  centroid: number;
}

export interface AmplitudeFeatures {
  envelope: number[];
  rms: number;
}

export interface AudioAnalysisResponse {
  frequency_features: FrequencyFeatures;
  amplitude_features: AmplitudeFeatures;
  similarity_score?: number;
}

export interface AudioQualityMetrics {
  duration: number;
  maxAmplitude: number;
  avgAmplitude: number;
  isClipping: boolean;
  isTooQuiet: boolean;
} 

File: backend/app/models/audio.py
from pydantic import BaseModel
from typing import List, Tuple

class FrequencyFeatures(BaseModel):
    fundamental: float
    spectrum: List[float]
    centroid: float

class AmplitudeFeatures(BaseModel):
    envelope: List[float]
    rms: float

class AudioFeatures(BaseModel):
    frequencyRange: Tuple[float, float]
    amplitudeRange: Tuple[float, float]
    durationRange: Tuple[float, float]
    centroid: float
    rms: float

class AudioAnalysisResponse(BaseModel):
    frequency_features: FrequencyFeatures
    amplitude_features: AmplitudeFeatures
    similarity_score: float | None = None 